tensorboard/uploader/proto/write_service.proto

```diff
@@ -287,15 +289,47 @@ message GetOrCreateBlobSequenceResponse {
   string blob_sequence_id = 1;
 }
 
+message GetBlobMetadataRequest {
+  // The ID of the BlobSequence of which this blob is a member.
+  string blob_sequence_id = 1;
+  // The position of this Blob within the BlobSequence
+  int64 index = 2;
+}
+
+message GetBlobMetadataResponse {
+  enum BlobState {
+    // Object state is unknown. This value should never be used; it is present
+    // only as a proto3 best practice.
+    // See https://developers.google.com/protocol-buffers/docs/proto3#enum
+    UNKNOWN = 0;
+
+    // Object is being written and not yet finalized.
+    UNFINALIZED = 1;
+
+    // Object is finalized.
+    CURRENT = 2;
+  }
+
+  // State of the object (still appending vs. complete).
+  BlobState object_state = 1;
+  // Size of the object in bytes.  This is the sum of the chunk sizes
+  // received from the stream so far.  In the response to the final chunk,
+  // this size should equal the total size of the blob.
+  int64 size = 2;
+  // crc32c of the blob data read so far, i.e. over the byte range [0, size).
+  fixed32 cumulative_crc32c = 3;
```

Comment:
Optional but IMO at this point maybe better to just call this `crc32c` (no cumulative) for consistency with `size` which is just `size` rather than `cumulative_size`.  I think there's only really one obvious interpretation of what the CRC and size refer to in the new structure, which is whatever amount of the blob currently exists on the server (since that's the whole point of this RPC).

Reply:
Done